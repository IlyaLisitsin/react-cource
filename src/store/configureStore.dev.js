/* eslint-disable */
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';

import reducers from '../reducers';
// import promiseMiddleware from 'redux-promise-middleware';
// import { routerMiddleware } from 'react-router-redux';
// import reducers from 'redux-modules';
//
// import { persistReducer } from 'redux-persist';
// import storage from 'redux-persist/lib/storage';
//
// const persistConfig = {
//   key: 'root',
//   storage,
//   whitelist: ['auth'],
// };
//
// const persistedReducer = persistReducer(persistConfig, reducers);
//
// let composeEnhancers = compose;


// export default function configureStore(history) {
//   const router = routerMiddleware(history);
//   const enhancer = composeEnhancers(applyMiddleware(thunk, promiseMiddleware(), router));
//   const store = createStore(persistedReducer, enhancer);
//   return store;
// }

const devtools = process.env.NODE_ENV === 'test'
    ? x => x /* eslint-disable no-underscore-dangle */
    : window.__REDUX_DEVTOOLS_EXTENSION__
    && window.__REDUX_DEVTOOLS_EXTENSION__();
/* eslint-enable no-underscore-dangle */

// const store = createStore(reducers, compose(window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), applyMiddleware(thunk)));
const appliedMiddlewares = applyMiddleware(thunk);
const store = createStore(
    reducers,
    compose(
        appliedMiddlewares,
        devtools
    ),

);

export default store;
