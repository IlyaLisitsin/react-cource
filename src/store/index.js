import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createBrowserHistory } from 'history';
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

import createRootReducer from '../reducers';


export const history = createBrowserHistory();
const appliedMiddlewares = applyMiddleware(thunk);

const isProd = process.env.NODE_ENV === 'production';
const isTest = process.env.NODE_ENV === 'test';

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, createRootReducer(history))

export default function configureStore(preloadedState) {
    const store = createStore(
        persistedReducer,
        preloadedState,
        isProd || isTest ?
            compose(
                appliedMiddlewares,
            ) :
            compose(
                appliedMiddlewares,
                window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(), // eslint-disable-line no-underscore-dangle
            ),

    );

    const persistor = persistStore(store);

    return { store, persistor };
}
