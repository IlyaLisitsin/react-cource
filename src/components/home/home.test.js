import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";

import Home from './home.component';
import store from "../../store";

test('renders Home correctly', () => {
    const tree = renderer.create(
        <Provider store={store}>
            <Home />
        </Provider>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});