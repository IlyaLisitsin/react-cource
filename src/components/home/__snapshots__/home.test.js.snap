// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
Array [
  <header>
    <div
      className="header-wrapper"
    >
      <h2>
        Find your movie
      </h2>
      <span
        className="ant-input-group-wrapper"
      >
        <span
          className="ant-input-wrapper ant-input-group"
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value=""
          />
          <span
            className="ant-input-group-addon"
          >
            <i
              aria-label="icon: rollback"
              className="anticon anticon-rollback"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="rollback"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M793 242H366v-74c0-6.7-7.7-10.4-12.9-6.3l-142 112a8 8 0 0 0 0 12.6l142 112c5.2 4.1 12.9.4 12.9-6.3v-74h415v470H175c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h618c35.3 0 64-28.7 64-64V306c0-35.3-28.7-64-64-64z"
                />
              </svg>
            </i>
          </span>
        </span>
      </span>
      <div
        className="header-bottom"
      >
        <div
          className="ant-radio-group ant-radio-group-outline"
        >
          <span
            className="search-by"
          >
            Search by
          </span>
          <label
            className="ant-radio-wrapper"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="title"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Title
            </span>
          </label>
          <label
            className="ant-radio-wrapper"
          >
            <span
              className="ant-radio"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-radio-input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="radio"
                value="genre"
              />
              <span
                className="ant-radio-inner"
              />
            </span>
            <span>
              Genre
            </span>
          </label>
        </div>
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="submit"
        >
          <span>
            Search
          </span>
        </button>
      </div>
    </div>
  </header>,
  <div
    className="movies-container-bar"
  >
    <div
      className="movies-found"
    >
      0
       movies found
    </div>
    <div>
      <span>
        Sort by:
      </span>
      <div
        className="ant-radio-group ant-radio-group-outline"
      >
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="release_date"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
          <span>
            release date
          </span>
        </label>
        <label
          className="ant-radio-wrapper"
        >
          <span
            className="ant-radio"
            style={Object {}}
          >
            <input
              checked={false}
              className="ant-radio-input"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              type="radio"
              value="vote_average"
            />
            <span
              className="ant-radio-inner"
            />
          </span>
          <span>
            rating
          </span>
        </label>
      </div>
    </div>
  </div>,
  <div
    className="movies-container"
  >
    <div
      className="no-films-found"
    >
      No films found
    </div>
  </div>,
]
`;
